packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -51,13 +91,18 @@ export default function UnsavedInnerBlocks( {
 		}
 	}, [ blocks ] );
 
-	// If the current inner blocks object is different in any way
-	// from the original inner blocks from the post content then the
-	// user has made changes to the inner blocks. At this point the inner
-	// blocks can be considered "dirty".
-	// We also make sure the current innerBlocks had a chance to be set.
-	const innerBlocksAreDirty =
-		!! originalBlocks.current && blocks !== originalBlocks.current;
+	const innerBlocksAreDirty = ! isDeepEqual(
+		originalBlocks.current,
+		blocks,
+		( prop, x ) => {
+			// skip inner blocks of page list during comparison as they
+			// are always controlled and may be updated async due to
+			// syncing with enitiy records.
+			if ( x?.name && prop === 'innerBlocks' ) {
+				return true;
+			}
+		}
+	);
```

Comment:
This approach now handles more scenarios. For example if you have the following _uncontrolled_ Nav block containing a _mix_ of Page List _and_ other block types. It will check all the blocks including Page List but just ignore the inner blocks of the Page List (only).

```
<!-- wp:navigation -->
<!-- wp:page-list /-->

<!-- wp:navigation-link {"label":"Laborum ducimus","type":"page","id":609,"url":"http://localhost:8888/laborum-ducimus-voluptatibus-officia-soluta-impedit/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:site-logo {"shouldSyncIcon":true} /-->

<!-- wp:navigation-link {"label":"Quas facere est ","type":"post","id":593,"url":"http://localhost:8888/quas-facere-est-doloribus-vel-consequuntur/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"isTopLevelLink":true} /-->
<!-- /wp:navigation -->
```

See related discussion here https://github.com/WordPress/gutenberg/pull/46223#discussion_r1036990506.